src/services/services.ts

```diff
@@ -2797,18 +2800,30 @@ namespace ts {
             return program.getOptionsDiagnostics().concat(program.getGlobalDiagnostics());
         }
 
-        /// Completion
-        function getCompletionEntryDisplayNameForSymbol(symbol: Symbol, target: ScriptTarget, performCharacterChecks: boolean): string {
-            let displayName = symbol.getName();
-            if (displayName) {
-                // If this is the default export, get the name of the declaration if it exists
-                if (displayName === "default") {
-                    let localSymbol = getLocalSymbolForExportDefault(symbol);
-                    if (localSymbol && localSymbol.name) {
-                        displayName = symbol.valueDeclaration.localSymbol.name;
-                    }
-                }
-
+        /**
+         * Get the name to be display in completion from a given symbol.
+         *
+         * @return undefined if the name is of external module otherwise a name with striped of any quote
+         */
+        function getCompletionEntryDisplayNameForSymbol(symbol: Symbol, target: ScriptTarget, performCharacterChecks: boolean, location: Node): string {

```

Comment:
You can't use ```getDeclaredName```. For example: ```"/u0042"``` will be store on its symbol as "B" while if you use ```getDeclaredName``` it will get the unicode escape sequences which is as expected from the function since it is a declared name.